### DISCUSSION TITLE
Import from tree html notes on old Keepnote

### DISCUSSION THREAD

**Discussion by joejac** [TriliumNext/Trilium#1933] [Q&A]
Thanks for sharing Trilium
I have my personal and research notes in Keepnote. I am use to it, but unfortunately is not maintained since 2012. I have about 880MB of notes containing text formatted in html plus some images per note. Each note is in a sub-directory containing **page.html** which is the note, a small **node.xml** file and the images of the note. How can I import all my notes into Trilium in an automatic way?
Link to Keepnote: [](http://keepnote.org/ )  I am using Version 0.7.8
Thanks in advance
Regards
joejac

**Comment by zadam** [comment]
Hi, Trilium can import the HTML files and will keep the directory structure (the whole import directory should be imported as ZIP file). But the keepnote's XML files won't be recognized.

If I were you I would probably try to write a script parsing the keepnote's XML, extracting the images etc. out of it and make a nice directory structure of all those HTML files, images, attachments etc so that it looks like a static website. Then it should be importable into trilium.

**Comment by joejac** [comment]
Thank you @zadam for your response and suggestion, unfortunately I am not a programmer. What I did is "Import into note" I select the full zip file and when it reaches probably  the end of notes the import process freezes, please see screen capture here [Import Freezes](https://ibb.co/6WYdLrM)  but the program is still running, I can even create a new note, I let it run for about 10 minutes and nothing happens, then I close the program and when I open it again about **one third of the notes where imported fine**, all text, its format and images, that is very good, but is missing about 60% of the notes. **¿Is there a limit of notes to import?** 

If I am able to import **all** my notes it would be great. I would need to manually  delete the xml files and complete the note title because it is truncated at directory level, that is not a problem I can do that, but the most important part is that text and images of notes are imported fine this is very good because other similar programs failed on this. So I just need that the import process does not freeze and its works **until the last note**. I appreciate any suggestion to complete the import process on all notes.
Best regards
joejac

**Reply by zadam** [reply]
Hi, freeze during import sounds like a bug. I can check that but in the mean time what should help is then splitting the ZIP archive into let's say 10 smaller ZIP files and importing them one by one?
